capten/agent/internal/crossplane/cluster_claims.go

```diff
@@ -422,3 +441,40 @@ func (h *ClusterClaimSyncHandler) syncClusterClaimsWithDB(clusterClaims []model.
 	}
 	return nil
 }
+
+func (h *ClusterClaimSyncHandler) checkForClusterUpdate(clusterCliam model.ClusterClaim) bool {
+
+	var statues []string
+	for _, condition := range clusterCliam.Status.Conditions {
+		statues = append(statues, condition.Status)
+	}
+
+	sameStatues := isSameStatues(clusterCache[clusterCliam.Metadata.Name].Statuses, statues)

```

Comment:
isSameStatues confusing, have all cluster status check one method.

ClusterCache change to ManagedClusterData
ManagedClusterData - add same attributes
 - add method match

Reply:
handling of ManagedClusterData is modified